### PR TITLE
Reload cached resources in runtime on file reimport

### LINKED ISSUE #93629: Texture does not update in running game if changed in project. Game needs to be restarted.
### Tested versions

4.3.beta2

### System information

Godot v4.3.beta2 - Windows 10.0.22631 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4070 Laptop GPU (NVIDIA; 31.0.15.4633) - AMD Ryzen 7 7840HS w/ Radeon 780M Graphics (16 Threads)

### Issue description

Textures don't update in running game if you change the texture file. So you have to restart the game. I would expect it to reload the texture when it changes.

### Steps to reproduce

https://github.com/godotengine/godot/assets/4955051/41bfffad-7867-4539-b118-e239d1a791f1

### Minimal reproduction project (MRP)

NA

### PR DESCRIPTION
**Pull request by YeldhamDev**
Closes #93629.

**Sponsored By:** 🐺 Lone Wolf Technology / 🍀 W4 Games.

### CODE REVIEW

**Review by clayjohn — APPROVED** [review]
The code looks good to me. But this should also get an approval from someone with more debugger knowledge

**Review by Faless — CHANGES_REQUESTED** [review]
Looks good overall, see my comment about the EditorFileSystem check.

I wonder if we should put the `reload_cached_files` behind the `live_debug` flag (i.e. synchronize scene changes), and thus also rename the message to `live_reload_cached_files` and move handling to the live editor.

File: editor/debugger/script_editor_debugger.cpp [hunk 1]
```diff
@@ -1819,6 +1825,10 @@ ScriptEditorDebugger::ScriptEditorDebugger() {
 
 	InspectorDock::get_inspector_singleton()->connect("object_id_selected", callable_mp(this, &ScriptEditorDebugger::_remote_object_selected));
 
+	if (EditorFileSystem::get_singleton()) {
+		EditorFileSystem::get_singleton()->connect("resources_reimported", callable_mp(this, &ScriptEditorDebugger::_resources_reimported));
+	}
```
[the remark this answers is not in the document]
Comment (YeldhamDev) [reply editor/debugger/script_editor_debugger.cpp]: I added that because further up there was also a check for this, so I just wanted to be sure. Removed.

**Comment by YeldhamDev** [comment]
> I wonder if we should put the `reload_cached_files` behind the `live_debug` flag (i.e. synchronize scene changes), and thus also rename the message to `live_reload_cached_files` and move handling to the live editor.

I thought about that, but decided against it because that's not really a "scene change", and more of an asset data modification.

**Review by Faless — APPROVED** [review]
LGTM

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**